test/tests/test_estimate_fee.rs

```diff
@@ -0,0 +1,200 @@
+#![feature(assert_matches)]
+
+use rpc_test::test_config::TestConfig;
+use starknet::{providers::{
+    jsonrpc::{HttpTransport, JsonRpcClient},
+    Provider, ProviderError, StarknetErrorWithMessage, MaybeUnknownErrorCode
+}, core::types::{BlockId, FieldElement, StarknetError, BlockTag, BroadcastedTransaction, BroadcastedInvokeTransaction}};
+use starknet::core::utils::get_selector_from_name;
+use url::Url;
+use std::assert_matches::assert_matches;
+//use starknet_providers::ProviderError::StarknetError as StarknetProviderError;
+
+const ACCOUNT_CONTRACT: &str = "";
+const TEST_CONTRACT_ADDRESS: &str = "";
+
+
+#[tokio::test]
+async fn fail_non_existing_block() {
+    let config = TestConfig::new("./secret.json").unwrap();
+    let deoxys = JsonRpcClient::new(HttpTransport::new(
+        Url::parse(&config.deoxys).unwrap()
+    ));
+
+    let ok_invoke_transaction = BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction {
+        max_fee: FieldElement::ZERO,
+        signature: vec![],
+        nonce: FieldElement::ZERO,
+        sender_address: FieldElement::from_hex_be(ACCOUNT_CONTRACT).unwrap(),
+        calldata: vec![
+            FieldElement::from_hex_be(TEST_CONTRACT_ADDRESS).unwrap(),
+            get_selector_from_name("sqrt").unwrap(),
+            FieldElement::from_hex_be("1").unwrap(),
+            FieldElement::from(81u8),
+        ],
+        is_query: true,
+    });
+
+    assert_matches!(
+        deoxys.estimate_fee(&vec![ok_invoke_transaction], BlockId::Hash(FieldElement::ZERO)).await,
```

Comment:
could you store both sides of the assert on a variable then reference it in the assert

Reply:
same everywhere else